RULES.md

```diff
@@ -0,0 +1,94 @@
+- Organization administration - ability to grant and remove rights, create and delete repositories etc.
+
+
+## Implementation
+
+### Products
+
+- Keeping track of the decomposition: Metadata in product.yml files, CODEOWNERS file
+- Area of effect: All PSPs that don’t have specific stakeholders, associated content and test scenarios.
+- Guidelines: Care needs to be taken when content requires a slightly different approach for different products - don’t copy paste, and also don’t change existing behavior for everybody (link to coding guidelines needed).
+
+
+### Product Specific Profiles (PSPs)
+
+- Keeping track of the decomposition: Metadata in profile yml files, CODEOWNERS file
+- Area of effect: Respective profile files
+- Guidelines: N/A
+
+
+### Product Independent Profiles (PIPs)
+
+- Keeping track of the decomposition: Metadata in control files. Any profile definitions that live as separate profile files are simply a product’s content.
+- Area of effect: Just control files
+- Guidelines: Let’s try and see whether the control file format helps us with effort sharing.
+
+
+### Build system
+
+- Keeping track of the decomposition: CODEOWNERS file
+- Area of effect: Everything that is directly or indirectly used during the content build process
+- Guidelines: We need strong protection against technical debt, and a framework to deal with deprecations or changes of behavior.
+
+
+### Shared Resources
+
+- Keeping track of the decomposition: CODEOWNERS file
+- Area of effect: Content that impacts more products.
+- Guidelines: The same as for the build system.
+
+
+### Tests
+
+- Keeping track of the decomposition: CODEOWNERS file
+- Area of effect: Test scenarios, test suite features, unit tests, gating tests
+- Guidelines: We need guidelines to preserve test robustness and the rate of tests usefulness.
+
+
+### Organizational administration
+
+- Keeping track of the decomposition: Maintainer file, readme?
+- Area of effect: Github org and project settings
+- Guidelines: Enter and exit conditions
+
+
+## Rights assignment
+
+### Individual PR approve and merge rights
+
+TLDR: 6 non-trivial PRs within 6 months => merge rights for 1 year since the last activity.
+
+Contributors can ask for merge rights on the mailing list or on Gitter.
+Although one can technically merge code, the code contribution guidelines still have to be obeyed.
+
+Recommended way to get merge rights: Anybody who submits six non-trivial PRs that get merged in a period no longer than six months are encouraged to ask for rights.
+A person can ask to be stripped of merge rights.
+If a person doesn’t actively contribute to the project for a period of 12 months, their merge rights are stripped. Active contribution consists of either non-trivial contributions of code, non-trivial reviews, or analyses of problems in issues.  Rights can be also revoked if coding style or conduct are consistently broken.
```

Comment:
Just curious how this is going to work. `ComplianceAsCode` is admittedly a different org than `OpenSCAP` but we still have 15+ repos under the CaC org. Are we going to scope contributors to particular projects? Do merge bits apply across projects? If `OpenSACAP/scap-workbench` were in this org, it might be hard to get 6 non-trivial PRs in 6 months to that repo (I tried while working at Canonical and only managed 4 with 2 failed -- and some of those 4 were trivial! :D) _and_ another 6 to `CaC/content` to have commit bits in both, but I could definitely see why the separation would be good. Would there be some threshold of contributions to be a "organization" member versus a "`CaC/content` contributor if that makes sense? Would the former retain merge commits longer than the latter? &c. 

Also, is this strictly removing the _merge_ bits or does it remove ACK'ing rights as well? E.g., assuming I get busy at HCP (likely) and don't have time here, it makes sense that I'd lose merge access, but would my review still count for +1 on PRs (allowing others to merge)? Or would it be +0? 

Just asking the hard questions ;-)

Reply:
This is a guideline only for the this (ComplianceAsCode/content) project. The process of getting commit rights for other projects are not defined - whoever wants them can ask on MLs.

As noted below, 6 PRs in 6 months is the "standard" way, but not the only one. One can say that it encourages people to create smaller PRs, or to reach out if they don't start from zero.

ACKing is not defined, and I would say that any PR review helps - if the change doesn't have to be reviewed by SMEs (e.g. it is a fix of Bash content or of OVAL) such reviews help to get PRs through faster. I think that we can work with informal "ACK power" - if one's username sounds familiar, then they are likely to have the ACK power.

Concerning organizations, it is ATM a very loose concept. We will iron it out based on feedback and needs of contributing organizations.

Reply:
There's two types of ACKs in GitHub: ACKs by "approved" members ([example](https://github.com/hashicorp/vault-selinux-policies/pull/8#pullrequestreview-903061554)) and ACKs by random community members ([example](https://github.com/hashicorp/terraform-provider-tls/pull/151#pullrequestreview-891610088)). Even though I don't have merge bits in either repository, I've been added to a list of PR reviewers on the former repo, so my review still "counts" as a +1 rather than a +0. This lets someone who _does_ have merge bits merge the PR without having to ACK it themselves (counts as an independent review in the case where PRs need to be ACKd before being merged).

I bring this up because these finer permissions might be useful to the project. SUSE &c could have many _mergeless_ reviewers (and this could be given fairly liberally), but fewer people with merge access.

Perhaps rather than removing the user from the repo entirely (after 6 months or whatever), they could be dropped to approvals only for the next period of time? 

My 2c.

Reply:
Edit to add (and in case anyone is wondering _how_ this is accomplished), apparently I had "write" access to the repository at the time of this discussion -- but due to branch protections, I lacked permission to merge.

Reply:
I think that we don't have to distinguish between authoritative approvals and merges - I wouldn't look at it as stripping the approval when the merge right expires, but rather as keeping the merge right as long as we think that there is a solid chance of getting a qualified review out of that person.
As the period of long inactivity passes, I would suggest to treat people as "random community members" again, and it should be easier to regain merge rights, e.g. by contributing reviews.